### ISSUE TITLE
Instrument view crash

### ISSUE THREAD

**Issue by AndreiSavici** [mantidproject/mantid#33260]
**Describe the bug**
Originally seen in StepScanAnalysis interface: after loading a file, user clicks on `Define` the pixel mask. Mantid crashes. The issue seems to be the python code that is calling the instrument view, in particular the `select_tab` function. One can reproduce this with a simple script as well. This used to work in 6.2.0, does not work in nightly or ORNL newer versions.

**To Reproduce**
In Mantid workbech, in the python interpreter, paste the following code (the file is in the standard test data)

```python
w=Load('CNCS_7860')
from mantidqt.widgets.instrumentview.api import get_instrumentview
instrument_view = get_instrumentview('w')
instrument_view.select_tab(2)
instrument_view.show_view()
```
press Enter.

Note that if you run line by line, one at a time, it seems to be OK :scream: 
Also, commenting out the select tab line will make the script work

**Platform/Version (please complete the following information):**
 - OS: RHEL7, Ubuntu 20.04
 - Mantid Version > 6.2.0

**Additional context**
Any fix in the script should be propagated to [StepScanAnalysys interface](https://github.com/mantidproject/mantid/blob/main/qt/scientific_interfaces/General/StepScan.cpp#L343-L348).

**Comment by martyngigg** [comment]
@thomashampson You've been looking at a crash there too. Could it be related?

**Comment by thomashampson** [comment]
This problem appears to go away when setting the default value `useThread = false` here:
https://github.com/mantidproject/mantid/blob/cc12a281eb2a83e659bfe03e72b70f91354519dc/qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h#L120

and is likely related to the instrument viewer crashing in Windows debug mode if it's closed while still loading.

**Comment by martyngigg** [comment]
> This problem appears to go away when setting the default value `useThread = false` here:
> 
> https://github.com/mantidproject/mantid/blob/cc12a281eb2a83e659bfe03e72b70f91354519dc/qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/InstrumentWidget.h#L120
> 
> and is likely related to the instrument viewer crashing in Windows debug mode if it's closed while still loading.

I'm highly supportive of us setting this to false by default as it doesn't sound like it's ready for production in threaded-mode yet?

**Comment by thomashampson** [comment]
Debugging in Windows, running the script gives this message box (this might be information that you're missing on Linux systems):
![image](https://user-images.githubusercontent.com/8058899/149369936-4091a930-ca33-4f71-8fae-14eba684ce7e.png)

**Comment by AndreiSavici** [comment]
@thomashampson @martyngigg See if the fix works for you as well

**Comment by martyngigg** [comment]
Fixed by #33349
